### PR TITLE
Update the compiler in the Linux/Mac toolset

### PR DESCRIPTION
**Pull request by agocke**
Updates the compiler used in the Linux and Mac toolsets and also eliminates the requirement to download the toolset as a ZIP and instead restores the necessary components over NuGet.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,50 +3,46 @@ OS_NAME = $(shell uname -s)
...
-RESTORE_SEMAPHORE_PATH = $(TOOLSET_PATH)/restore.semaphore
+RESTORE_SEMAPHORE_PATH = $(BINARIES_PATH)/restore.semaphore
 BOOTSTRAP_PATH = $(BINARIES_PATH)/Bootstrap
 BUILD_LOG_PATH =
 HOME_DIR = $(shell cd ~ && pwd)
 DOTNET_VERSION = 1.0.0-preview2-002911
 NUGET_VERSION = 3.5.0-beta2
 NUGET_EXE = $(shell pwd)/nuget.exe
 
-MSBUILD_ADDITIONALARGS := /v:m /fl /fileloggerparameters:Verbosity=normal /p:Configuration=$(BUILD_CONFIGURATION)
```
Comment (jaredpar) [inline Makefile]: Can't build in parallel anymore?

File: build/MSBuildToolset/csc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+THISDIR=$(dirname $0)
+
+chmod +x $THISDIR/corerun 2>/dev/null
+$THISDIR/corerun $THISDIR/csc.exe "$@"
```
Comment (jaredpar) [inline build/MSBuildToolset/csc]: This line implies that csc.exe is going to be written to this directory.  That feels wrong as it's build deliberately modifying the source tree.

File: build/MSBuildToolset/vbc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+THISDIR=$(dirname $0)
+
+chmod +x $THISDIR/corerun 2>/dev/null
+$THISDIR/corerun $THISDIR/vbc.exe "$@"
```
Comment (jaredpar) [inline build/MSBuildToolset/vbc]: Same

File: build/Targets/Imports.targets [hunk 4]
```diff
@@ -155,6 +155,7 @@
           <PropertyGroup>
             <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)..\Strong Name Keys\RoslynInternalKey.Private.snk</AssemblyOriginatorKeyFile>
             <DelaySign>false</DelaySign>
+            <PublicSign>true</PublicSign>
```
Comment (jaredpar) [inline build/Targets/Imports.targets]: This feels wrong.  This PropertyGroup is only hit when we are fully signing binaries, not public signing.

File: Makefile (hunk 1 above)
Comment (agocke) [inline Makefile]: Woops, I was debugging and forgot to delete that change. `/v:m` is minimal verbosity, not parallel build.

File: build/MSBuildToolset/csc (hunk 2 above)
Comment (agocke) [inline build/MSBuildToolset/csc]: No, that file is copied to the output directory. The build outputs to `Binaries/toolset`

**Comment by jaredpar** [comment]
Ignore the  windows_build_correctness_prtest job.  Jenkins shouldn't have added it yet.

**Comment by agocke** [comment]
test mac please

**Comment by jaredpar** [comment]
Looks like the close / open fixed the jenkins pointing at the wrong jobs problem.

File: Makefile [hunk 5]
```diff
@@ -77,7 +77,8 @@ toolset: $(TOOLSET_PATH)
 $(TOOLSET_PATH): $(BINARIES_PATH)/dotnet-cli
 	pushd $(TOOLSET_SRC_PATH) ; \
 	$(BINARIES_PATH)/dotnet-cli/dotnet restore && \
-	$(BINARIES_PATH)/dotnet-cli/dotnet publish -o $(TOOLSET_PATH)
+	$(BINARIES_PATH)/dotnet-cli/dotnet publish -o $(TOOLSET_PATH) && \
+	sed -i -e 's/Microsoft.CSharp.Targets/Microsoft.CSharp.targets/g' $(TOOLSET_PATH)/Extensions/Microsoft/Portable/v5.0/Microsoft.Portable.CSharp.targets
```
Comment (jaredpar) [inline Makefile]: Seems like we should fix.  Is there a tracking bug on us or the appropriate owner?

Comment (agocke) [inline Makefile]: @jaredpar Tracking bug was here: https://github.com/dotnet/roslyn/issues/9641

But the NuGet package we created to hold the portable targets hasn't been updated. I'm not sure if that update is even planned, given the new track for portable projects.

Comment (jaredpar) [inline Makefile]: Please reference that bug in this code.

File: src/Tools/RepoUtil/ChangeCommand.cs [hunk 6]
```diff
@@ -154,6 +154,13 @@ private List<NuGetPackageChange> CalculateChanges(IEnumerable<NuGetPackage> pack
             var map = _repoData
                 .FloatingPackages
                 .ToDictionary(x => x.Name, Constants.NugetPackageNameComparer);
+            foreach (var @fixed in _repoData.FixedPackages)
+            {
+                if (@fixed.GenerateNameOpt != null)
+                {
+                    map.Add(@fixed.GenerateNameOpt, @fixed.Package);
+                }
+            }
```
Comment (jaredpar) [inline src/Tools/RepoUtil/ChangeCommand.cs]: Did you test that change actually works on fixed packages here?  I can't remember if I accounted for this or not when I wrote the code.

Comment (agocke) [inline src/Tools/RepoUtil/ChangeCommand.cs]: Yup, the dependencies.props change was created using my modifications.

File: src/Tools/RepoUtil/GenerateUtil.cs [hunk 7]
```diff
@@ -24,11 +24,18 @@ internal static class GenerateUtil
...
-            // possible for two versions to exist for the same package.  Take for example System.Collections.Immuatble
-            // which is both fixed and floating in Roslyn.
+            // Only fixed packages with 'generate names' are included in generated output. Doing
+            // otherwise would create conflicts because it's possible for two versions to exist
+            // for the same package.  Take for example System.Collections.Immuatble which is both
+            // fixed and floating in Roslyn.
+            var fixedWithGenerate = repoData
+                .FixedPackages
+                .Where(x => x.GenerateNameOpt != null)
+                .Select(x => new NuGetPackage(x.GenerateNameOpt, x.Package.Version));
```
Comment (jaredpar) [inline src/Tools/RepoUtil/GenerateUtil.cs]: I'm nervous about this approach.  It's creating a NuGetPackage object which not correct: a fake name paired with a version.  Consider a case where the fake name actually points to a real package.  Could cause issues down the line. 

I think we should extend NuGetpPackage to have the concept of GenerateName.  Defaults to package name but can be overriden.

File: src/Tools/RepoUtil/ChangeCommand.cs (hunk 6 above)
Comment (agocke) [inline src/Tools/RepoUtil/ChangeCommand.cs]: Woops, I was reading the wrong line. You're right, I need to test that feature more.

Comment (agocke) [inline src/Tools/RepoUtil/ChangeCommand.cs]: On second thought, I'm just going to remove that functionality. It's not necessary.

File: src/Tools/RepoUtil/GenerateUtil.cs (hunk 7 above)
Comment (agocke) [inline src/Tools/RepoUtil/GenerateUtil.cs]: @jaredpar Cool, I'll make the change.

**Comment by agocke** [comment]
test windows_release_unit32_prtest please

**Pull request merged**